inc/sort.php

```diff
@@ -0,0 +1,188 @@
+ * It uses a collator-based sort, or ksort() with flags SORT_NATURAL and SORT_FLAG_CASE as a fallback.
+ *
+ * @param array $array The input array.
+ * @return bool Returns true on success or false on failure.
+ *
+ * @author Moisés Braga Ribeiro <moisesbr@gmail.com>
+ */
+function intl_ksort(&$array)
+{
+    $collator = _get_collator();
+    if (isset($collator)) {
+        return uksort($array, array($collator, 'compare'));
+    } else {
+        return ksort($array, SORT_NATURAL | SORT_FLAG_CASE);
+    }
+}
+
+/**
+ * Drop-in replacement for asort(), natsort() and natcasesort().
+ * It uses a collator-based sort, or asort() with flags SORT_NATURAL and SORT_FLAG_CASE as a fallback.
+ *
+ * @param array $array The input array.
+ * @return bool Returns true on success or false on failure.
+ *
+ * @author Moisés Braga Ribeiro <moisesbr@gmail.com>
+ */
+function intl_asort(&$array)
+{
+    $collator = _get_collator();
+    if (isset($collator)) {
+        return $collator->asort($array);
+    } else {
+        return asort($array, SORT_NATURAL | SORT_FLAG_CASE);
+    }
+}
+
+/**
+ * Drop-in replacement for asort(), natsort() and natcasesort() when the parameter is an array of filenames.
+ * Filenames may not be equal to page names, depending on the setting in $conf['fnencode'],
+ * so the correct behavior is to sort page names and reflect this sorting in the filename array.
+ *
+ * @param array $array The input array.
+ * @return bool Returns true on success or false on failure.
+ *
+ * @author Moisés Braga Ribeiro <moisesbr@gmail.com>
+ */
+function intl_asortFN(&$array)
+{
+    $collator = _get_collator();
+    if (isset($collator)) {
+        return uasort($array, '_sort_filenames_with_collator');
+    } else {
+        return uasort($array, '_sort_filenames_without_collator');
+    }
+}
+
+/**
+ * Collator-based string comparison for filenames.
+ * The filenames are converted to page names with utf8_decodeFN() before the comparison.
+ * 
+ * @param string $fn1 The first filename.
+ * @param string $fn2 The second filename.
+ * @return int Returns < 0 if $fn1 is less than $fn2; > 0 if $fn1 is greater than $fn2, and 0 if they are equal.
+ *
+ * @author Moisés Braga Ribeiro <moisesbr@gmail.com>
+ */
+function _sort_filenames_with_collator($fn1, $fn2)
+{
+    $collator = _get_collator();
+    return $collator->compare(utf8_decodeFN($fn1), utf8_decodeFN($fn2));
+}
+
+/**
+ * Fallback string comparison for filenames, using strnatcasecmp().
+ * The filenames are converted to page names with utf8_decodeFN() before the comparison.
+ * 
+ * @param string $fn1 The first filename.
+ * @param string $fn2 The second filename.
+ * @return int Returns < 0 if $fn1 is less than $fn2; > 0 if $fn1 is greater than $fn2, and 0 if they are equal.
+ *
+ * @author Moisés Braga Ribeiro <moisesbr@gmail.com>
+ */
+function _sort_filenames_without_collator($fn1, $fn2)
+{
+    return strnatcasecmp(utf8_decodeFN($fn1), utf8_decodeFN($fn2));
+}
```

Comment:
I would prefer to have these function wrapped into a namespaced class `\dokuwiki\Utf8\Sort` similar to the other UTF-8 classes we have. We could then call `Sort::ksort()` instead of `intl_sort()`